azkaban-common/src/test/java/azkaban/utils/TestUtils.java

```diff
@@ -26,6 +27,10 @@
 import java.io.File;
 import java.io.IOException;
 import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionFactory;
+import org.hamcrest.Matcher;
```

Comment:
Does assertj provide similar functionalities? We favor assertj over hamcrest per the dev guide.

Reply:
https://blog.jayway.com/2014/04/23/java-8-and-assertj-support-in-awaitility-1-6-0/

https://solidsoft.wordpress.com/2014/04/29/using-assertj-and-awaitility-together-thanks-to-java-8-and-lambdas/

Reply:
I don't understand how to make that work. This doesn't compile:

```java
    Awaitility.await().until(() ->
        org.assertj.core.api.Assertions.assertThat(getFlowStatus(flow))
            .matches(Status::isStatusFinished)
    );
```

Also tried to take the example from the second [link](https://blog.jayway.com/2014/04/23/java-8-and-assertj-support-in-awaitility-1-6-0/) but it doesn't compile either:

```java
    Awaitility.await().until( () -> org.assertj.core.api.Assertions.assertThat(1).isEqualTo(1) );
```

Unless you know how to use this, I propose we go with the "standard" awaitility+hamcrest combo.

Reply:
Below example works for me.

```java
  @Test
  public void test_wait() {
    Awaitility.await().untilAsserted(() -> assertThat(1).isEqualTo(1));
  }
```
See
https://github.com/awaitility/awaitility/wiki/Usage#version-3x-and-above

Reply:
Hmm, I just came across this, let me try it:
https://github.com/awaitility/awaitility/wiki/Usage#version-3x-and-above-1

Reply:
Yeah, this works:

```java
    TestUtils.await().untilAsserted(() -> assertThat(getFlowStatus(flow))
        .matches(Status::isStatusFinished, "a finished status"));
```